idb/postgres/internal/writer/writer.go

```diff
@@ -159,7 +160,7 @@ func getSigTypeDeltas(payset []transactions.SignedTxnInBlock) (map[basics.Addres
 	res := make(map[basics.Address]sigTypeDelta, len(payset))
 
 	for i := range payset {
-		if payset[i].Txn.RekeyTo == (basics.Address{}) {
+		if payset[i].Txn.RekeyTo == (basics.Address{}) && payset[i].Txn.Type != protocol.StateProofTx {
```

Comment:
Did you consider pushing this logic down into `idb.SignatureType`? It could return `None`.

Maybe it's fine like this, makes it clear that there is a special case relating to state proofs.

Reply:
Yeah I actually was trying fit it into that function first, but it returns `(SigType, error)` where `SigType("")` is always an error because we essentially have an unsigned Txn in the block. It just seemed to be more confusing to either have an empty SigType that isn't an error in there, or one that is so I just decided to bypass checking the sig type altogether for StateProofs.